ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -626,41 +636,41 @@ private object OracleQueries extends Queries {
   )(implicit log: LogHandler, pas: Put[Array[String]]): ConnectionIO[Int] = {
     println("insert contracts")
     println(dbcs)
-    val r = Update[(String, String, SurrogateTpId, JsValue, JsValue, String)](
+    val r = Update[(String, SurrogateTpId, JsValue, JsValue, String)](
       """
-        MERGE INTO contract USING (SELECT 1 FROM DUAL) ON (contract_id = ?)
-        WHEN NOT MATCHED THEN INSERT (contract_id, tpid, key, payload, agreement_text)
+        INSERT /*+ ignore_row_on_dupkey_index(contract(contract_id)) */
+        INTO contract (contract_id, tpid, key, payload, agreement_text)
         VALUES (?, ?, ?, ?, ?)
       """,
       logHandler0 = log,
     ).updateMany(
       dbcs
         .map { c =>
 //          println(c)
-          (c.contractId, c.contractId, c.templateId, c.key, c.payload, c.agreementText)
+          (c.contractId, c.templateId, c.key, c.payload, c.agreementText)
         }
     )
     println("inserted")
     import cats.syntax.foldable._, cats.instances.vector._
-    val r2 = Update[(String, String, String, String)](
+    val r2 = Update[(String, String)](
       """
-        MERGE INTO signatories USING (SELECT 1 FROM DUAL) ON (contract_id = ? AND party = ?)
-        WHEN NOT MATCHED THEN INSERT (contract_id, party)
+        INSERT /*+ ignore_row_on_dupkey_index(signatories(contract_id, party)) */
```

Comment:
I'm not sure I follow what `/*` is. It looks like a C-style comment but it's embedded in a multi-line string that ultimately becomes a SQL, how is this parsed? If by any chance this is parsed as a Scala comment, should it stay? If not, I don't understand it.

Reply:
This is oracle magic fro `on conflict do nothing`. Yes really, magic comments are how this work …

Reply:
Ok.

Reply:
Thanks.